.github/workflows/nightly-build.yml

```diff
@@ -0,0 +1,68 @@
+name: Nightly Build
+
+on:
+  workflow_dispatch: # to be able to run it manually
+  schedule:
+    - cron: '0 0 * * *' # once a day
+
+jobs:
+  evaluateChanges:
+    name: Evaluate changes for run or skipping nightly build
+    runs-on: ubuntu-latest
+    outputs:
+      SHOULD_BUILD: ${{ steps.check.outputs.shouldBuild }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: Check
+        id: check
+        run: |
+          if [[ $(git rev-list --after="24 hours" main) ]]; then
+            echo ::set-output name=shouldBuild::true
+          else
+            echo ::set-output name=shouldBuild::false
+          fi
+
+
+  build:
+    name: Generate Nightly Build
+    runs-on: ubuntu-latest
+    needs: evaluateChanges
+    if: ${{ needs.evaluateChanges.outputs.SHOULD_BUILD == 'true' }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Build and send apks to telegram
+        run: |
+          for TASK in assembleWebsiteNightly; do
+            ./gradlew $TASK
+            apk_path="$(find . -type f -iname *.apk)"
+            echo $apk_path
+            curl https://api.telegram.org/bot${{ secrets.TELEGRAM_TOKEN }}/sendDocument \
+              -F chat_id="\${{ secrets.TELEGRAM_CHAT_ID }}" \
+              -F "caption=Size: $(ls -l --block-size=K "$apk_path" | awk '{ print $5 }')" \
+              -F parse_mode=HTML \
+              -F document=@"$apk_path"
+            rm "$apk_path"
+          done
+
+      - name: Update the runtime dependencies report
+        run: |
+          ./gradlew app:dependencies --configuration websiteReleaseRuntimeClasspath | grep -e '--- ' > app/runtime-dependencies-report.txt
+
+      - name: Add and commit the report
+        uses: EndBug/add-and-commit@2d7c2e89fcb436f8e139cc80813bd030bdd40c4b
+        with:
+          default_author: github_actions
+          message: 'Update the runtime dependencies report'
+          add: app/*.txt
```

Comment:
I don't see any use in commiting this deps file every nightly for this project, tbh. Do these dep files come in handy in anyway for a project of Rethink's size?

Reply:
I thought this is useful because some people may care about exactly what dependencies goes into apps that they are using, I can remove it if you want or change the frequency of the update or even make it manual

Reply:
Gotcha  I feel, this could be commited everytime we `git tag` instead.